### ISSUE TITLE
Transformer train loss leads to Nan

### ISSUE THREAD

**Issue by hiyaroy12** [raywzy/ICT#10]
Hi, @raywzy 

I am trying to train the model on ImageNet with the following setting:
--data_path /ILSVRC2012/train --validation_path /ILSVRC2012/val --mask_path /PConv-Keras/irregular_mask/train_mask --batch_size 32 --train_epoch 100 --nodes 1 --gpus 8 --node_rank 0 --n_layer 35 --n_embd 1024 --n_head 8 --GELU_2 --image_size 32 --use_ImageFolder

But I am getting Nan for train and test loss (screenshot attached).
<img width="455" alt="ImageNet" src="https://user-images.githubusercontent.com/45819628/134884974-a324a74c-7426-4631-bd93-3d6d83812da8.png">

This is happening for smaller datasets like Paris streetview (#train_images: 14900, #test_images 100) as well (screenshot attached).
![Paris_streetview](https://user-images.githubusercontent.com/45819628/134885379-f93d7af5-2e03-47a2-abfa-a0cdbaa9afe0.png)

Any suggestions on how to fix this issue?

**Comment by raywzy** [comment]
Hi Hiya,
  I have not met such cases. Could you please try `--random_stroke` option and see what will happen?
